packages/worker/src/workflow/vm.ts

```diff
@@ -256,7 +255,27 @@ export class VMWorkflowCreator implements WorkflowCreator {
       });
     }
 
-    return new this(script, sourceMap, isolateExecutionTimeoutMs) as InstanceType<T>;
+    return new this(script, isolateExecutionTimeoutMs) as InstanceType<T>;
+  }
+
+  private static async makeScriptWithSourceMap(code: string) {
+    const sourceMappingUrlDataRegex = /\s*\n[/][/][#]\s+sourceMappingURL=data:(?:[^,]*),([^\s]+)\s*$/m;
+    const sourceMapMatcher = code.match(sourceMappingUrlDataRegex);
+
+    if (sourceMapMatcher) {
+      const sourceMap = Buffer.from(sourceMapMatcher[1], 'base64').toString();
+      const sourceMapParsed: RawSourceMap = JSON.parse(sourceMap);
+
+      const filename = `/temporal/${sourceMapParsed.file}`;
+      const script = new vm.Script(code, { filename });
+      const sourceMapConsumer = await new SourceMapConsumer(sourceMapParsed);
+
+      return { script, filename, sourceMapConsumer };
+    } else {
+      const filename = `/temporal/workflowbundle-${uuid4()}.js`;
+      const script = new vm.Script(code, { filename });
+      return { script, filename, sourceMapConsumer: undefined };
```

Comment:
When would you expect this to happen? Wondering if we should emit an error here.

Reply:
I don’t expect this to happen if the user has correctly followed the bundling process, except for bundles that will have been produced with 1.0.1 and will be run on 1.0.2 without being rebundled (Which isn’t safe anyway…)

But I still need to deal with this case somehow. I’d be willing though to go with a throw new Error(…) if you are ok with it

Reply:
In that case it could make sense to extract this in the main Node.js thread so the error could be caught by the user (would also save some (insignificant amount of?) work because right now we run this code once per worker thread.

Reply:
That's actually something I was considering... I'd probably move this to `getOrCreateBundle()`. What I'm unsure is what exactly should then be transferred to threads (ie. through `postMessage`)... Would it be reasonable (performance, memory...) to pass `sourceMapParsed` through the messaging port?

Reply:
The raw source map is transferrable, it's just a plain JS object.